### PR TITLE
Refactor/reenable permission in menu

### PR DESCRIPTION
**Pull request by soofstad**
> [!NOTE] 
> Depends on https://github.com/equinor/data-modelling-storage-service/pull/739

## What does this pull request change?
- Update the generated API
- Add back the Explorer context menu for "Modify ACL"
- Fix some styling of the dialog

Screenshot_20231228_093131

## Why is this pull request needed?
- Users need to be able to control access

### CODE REVIEW

**Review by henrikbossart — COMMENTED** [review]
This seems fine, but I'm holding off on approving until the PR it depends on is merged, so this can be tested. :+1:

**Review by eoaksnes — APPROVED** [review]

**Comment by soofstad** [comment]
Do you want to test it first @henrikbossart ?

**Review by henrikbossart — APPROVED** [review]

**Pull request merged**
